CI: Smoke test / conformance-test:

The `Smoke test / conformance-test` is broken on master: https://github.com/cilium/cilium/actions?query=workflow%3A%22Smoke+test%22. E.g., https://github.com/cilium/cilium/runs/1883726775?check_suite_focus=true#step:9:245.

It fails with the following error in `Check prometheus metrics`:
```
# github.com/prometheus/prometheus/cmd/promtool
go/src/github.com/prometheus/prometheus/cmd/promtool/main.go:645:35: not enough arguments in call to api.LabelValues
	have (context.Context, string, time.Time, time.Time)
	want (context.Context, string, []string, time.Time, time.Time)
Error: Process completed with exit code 2.
```

--- a/.github/workflows/smoke-test.yaml
+++ b/.github/workflows/smoke-test.yaml
@@ -134,7 +134,7 @@ jobs:
           cilium_pod=$(kubectl -n kube-system get po -o name --field-selector=status.phase==Running -l 'k8s-app=cilium' -o jsonpath='{.items[0].metadata.name}' )
           kubectl -n kube-system exec $cilium_pod -- sh -c "apt update && apt install curl -y"
           kubectl -n kube-system exec $cilium_pod -- curl http://localhost:9090/metrics > metrics.prom
-          go get -u github.com/prometheus/prometheus/cmd/promtool
+          GO111MODULE=on go get github.com/prometheus/prometheus/cmd/promtool@e4487274853c587717006eeda8804e597d120340 # This is commit hash for v2.24.1
           cat metrics.prom | $HOME/go/bin/promtool check metrics
 
       - name: Capture cilium-sysdump

